### DISCUSSION TITLE
what is  reset/password API requestBody

### DISCUSSION THREAD

**Discussion by ParthaPratima** [Sunbird-Ed/Community#471] [Bugs] [answered]
error : com.fasterxml.jackson.databind.exc.MismatchedInputException: No content to map due to end-of-input
 at [Source: (String)""; line: 1, column: 0]

at :: String response = HttpClientUtil.get(url, getHeaders(context), context);  on this code

  public static boolean isUserAccountDisabled(String userId, RequestContext context)
      throws Exception {
    String url =
        ProjectUtil.getConfigValue(JsonKey.SUNBIRD_SSO_LB_IP)
            + "/auth/admin/realms/"
            + ProjectUtil.getConfigValue(JsonKey.SUNBIRD_SSO_RELAM)
            + "/attack-detection/brute-force/users/"
            + fedUserPrefix
            + userId;
    String response = HttpClientUtil.get(url, getHeaders(context), context);
    logger.info(context, "KeycloakBruteForceAttackUtil:getUserStatus: Response = " + response);
    Map<String, Object> attackStatus = new ObjectMapper().readValue(response, Map.class);
    boolean isDisabled = ((boolean) attackStatus.get("disabled"));
    if (isDisabled) {
      logger.info(context, "check attack detection for userId : " + userId + ", " + attackStatus);
    }
    return isDisabled;
  }

Request Body :: {
 "request":{ 
"type":"email",
"key":"abc.pm@gmail.com",
"userId":"eae2125e-80f4-49dd-8894-e30e5347a235"
  }
}

**Comment by Abhign** [comment]
@vinukumar-vs @rajeevsathish Can you check this ticket?

**Comment by vinukumar-vs** [comment]
@reshmi-nair @rhwarrier Can you please tag the right API documentation for this

@Krishnaj20

**Reply by BharathwajShankar** [reply]
Hi @ParthaPratima if possible could you please share the complete stack trace details here, Also please make sure you have configured the Keycloak-related environment variables correctly along with the local setup of keycloak for this API to work.

**Reply by ParthaPratima** [reply]
yes i configured the keyclaok env
& the stack trace details ::   Reset-Password api  generate a post req for generate link for update password to keyclaok  having 403 error
--------------------------------------------------------------------------------------------------------------------------------------------------------------

POST  :  http://localhost:8080/auth/realms/<realm name>/get-required-action-link

{
    "redirectUri": "https://dev.sunbirded.org/resources",
    "clientId": "<client-name>",
    "expirationInSecs": "155520000",
    "userName": "<user_name>",
    "requiredAction": "UPDATE_PASSWORD"
}

ERROR:: 
{
    "errorMessage": "Does not have realm admin role."
}

**Reply by ParthaPratima** [reply]
> @reshmi-nair @rhwarrier Can you please tag the right API documentation for this
> 
> @Krishnaj20

For the Reset-Password api there is no sunbird api documention  present there

**Reply by reshmi-nair** [reply]
@ParthaPratima Please check whether below fields are configured in userorg service environment.

sunbird_sso_url: url for keycloak server
sunbird_sso_realm: keycloak realm name
sunbird_ssorname: keycloak _username: keycloak user name
sunbird_sso_password: keycloak password
sunbird_sso_client_id: key cloak client id
sunbird_sso_client_secret : keycloak client secret

Sample pwd reset api call to userorg service after otp verfication.
/user/v1/password/reset api 
{
	"request": {
		"type": "email",
		"key": "<emailid>",
		"userId": "<useid>",
		"reqData": "{\"iv\":\"4c6b802dd9255f79917cad9d4f2b0ed7a\",\"encryptedData\":\"1427443f81af4edb37cf5614501c20fb1670bdae8791ccbfdabc17558e260fc6d744c2f94b53c29a3bed46f795e1c802a51e3560e80b6c1b3c88d568e1d94f281bbe4d4bddeeed7f35461be467ecd3ec9adb2529e866688407139f4bd8faf6c01\"}"
	}
}
This api response will give the keycloak password reset url. 
sample url:
/auth/realms/sunbird/login-actions/required-action?execution=UPDATE_PASSWORD&client_id=<clientid>&tab_id=E5GtSHoRxTk

So make sure you have set the above environment variables, so that you can get the proper reset pwd link.

**Reply by reshmi-nair** [reply, marked as answer]
@ParthaPratima Please refer to the thread [LernDiscussions #111](https://github.com/orgs/Sunbird-Lern/discussions/111) and lets use it for further followups on same issue. Lets this thread and also [https://github.com/orgs/Sunbird-Ed/discussions/477](https://github.com/orgs/Sunbird-Ed/discussions/477) because these are duplicates for same issue.

**Reply by ParthaPratima** [reply]
ok

**Reply by Krishnaj20** [reply]
@ParthaPratima - since the discussion will now be tracked in Lern I'll mark this thread as answered. Hope that's OK with you.
